src/main/java/com/minecolonies/coremod/util/StructureWrapper.java

```diff
@@ -168,6 +180,57 @@ private void removeStructure(@NotNull final BlockPos pos)
         }
     }
 
+    /**
+     * Remove a structure surrounded by water from the world.
+     * Places water or Air, depending on the surrounding
+     *
+     * @param pos coordinates
+     */
+    private void removeStructureInWater(@NotNull final BlockPos pos)
+    {
+        setLocalPosition(pos);
+
+        // Find the waterlevel
+        int waterLevel = 0;
+        for (int j = structure.getHeight(); j > 0 && waterLevel == 0; j--)
+        {
+            for (int k = 0; k < structure.getLength() * 2; k++)
+            {
+                for (int i = 0; i < structure.getWidth() * 2; i++)
```

Comment:
This looks amazingly expensive though, going by twice the length and twice the width. 
Especially because we can be sure that at structure.getLength() + 1 there should be some sort of water already.

Since the structures can be a few hundred blocks high it also makes more sense to use binary search.

Reply:
ye you're right its pretty inefficient, gonna optimize it. Doesnt get called often though